docs/how-to/migrate-v1-v2.rst

```diff
@@ -0,0 +1,131 @@
+Migrate Charmed MLflow Version 1 to Version 2
+=====================================================
+
+This guide shows how to migrate Charmed MLflow version 1 to version 2. This guide assumes you are running the old Charmed MLflow stack version 1, which runs with MariaDB. With MLflow version 2, we only support the MySQL integration. This guide outlines how to move data from MariaDB to MySQL and how to migrate data from version 1 to version 2.1.1. Data from the object store doesn't need to be migrated.
+
+Prerequisites
+-------------
+
+This guide assumes the following:
+
+#. You have deployed MLflow version 1 with MariaDB, MLflow server version 1.x, and MinIO.
+#. You have CLI access to the machine where the Juju controller is deployed (all commands will be executed from there).
+
+MariaDB Backup
+--------------
+
+Install the ``mysqldump`` command:
+
+.. code-block:: bash
+
+   sudo apt update
+   sudo apt install mysql-client
+
+Backup the MariaDB database with the following command:
+
+.. code-block:: bash
+
+   mysqldump --host=<mariadb-charm-ip-address> --user=root --password=root --column-statistics=0 --databases database > mlflow-db.sql
+
+Deploy MySQL Charm
+-------------------
+
+Deploy the MySQL charm, which is needed for MLflow v2:
+
+.. code-block:: bash
+
+   juju deploy mysql-k8s --channel 8.0/beta --series jammy --trust
+
+.. note:: For MLflow version ``v.2.1``, we deploy the 8.0/beta version of the charm. You may deploy a more up to date version in your case.
+
+Please wait until the charm goes to active in ``juju status``. Then run the following command to get the password for MySQL:
+
+.. code-block:: bash
+
+   juju run-action mysql-k8s/0 get-password --wait
+
+Adjust the Database Backup
+--------------------------
+
+Rename the database from ``database`` (used in MariaDB) to ``mlflow`` (used in MySQL):
+
+.. code-block:: bash
+
+   sed 's/`database`/`mlflow`/g' mlflow-db.sql > mlflow-db-updated.sql
+
+Rename one duplicate constraint as MySQL does not allow that:
```

Comment:
So in case of moving from maria to mysql there are only two occurrences of `CONSTRAINT_1` in the dump (I tested with real usage data). The thing is that there are two tables with that constraint. Because of this the sed is sufficient. Of course if there are more coppies each one needs to be renamed to sth unique.

Reply:
OK, please review my updated text